### PR TITLE
Done the solution for task.

### PR DESCRIPTION
**Pull request by oleksandrCod**
Also have hardCode(

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,61 @@
...
+        }
+        String formatedData = getFormattedData(builder.toString());
+        File managedData = new File(toFileName);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(managedData))) {
+            for (int i = 0; i < formatedData.length(); i++) {
+                writer.write(formatedData.charAt(i));
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't write data to file " + managedData, exception);
+        }
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:33]: Extract `read` and `write` operations to the separate methods.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,61 @@
...
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String regex = " ";
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:14]: `regex` should be constant

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,61 @@
...
+        File readFile = new File(fromFileName);
+        StringBuilder builder = new StringBuilder();
+        String value;
+        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
+            while ((value = reader.readLine()) != null) {
+                builder.append(value).append(regex);
+            }
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:21]:
```suggestion
        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
            while (reader.ready()) {
                builder.append(reader.readLine()).append(regex);
            }
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,61 @@
...
+    }
 
+    public String getFormattedData(String data) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+        int result;
+        String regex = " ";
+        final String buy = "buy";
+        final String supply = "supply";
+        final String resultName = "result";
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:43]: Introduce as constants.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,61 @@
...
+        StringBuilder builder = new StringBuilder();
+        String[] arr = data.split(regex);
+        for (String tips : arr) {
+            int intValue = parseInt(tips.substring(tips.indexOf(",") + 1));
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:47]: Introduce as constant.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,61 @@
...
+            if (tips.contains("buy")) {
+                totalBuy += intValue;
+            } else if (tips.contains("supply")) {
+                totalSupply += intValue;
+            }
+        }
+        result = totalSupply - totalBuy;
+        builder.append(supply).append(",").append(totalSupply).append(System.lineSeparator());
+        builder.append(buy).append(",").append(totalBuy).append(System.lineSeparator());
+        builder.append(resultName).append(",").append(result);
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:57]:
```suggestion
        builder.append(supply).append(",").append(totalSupply).append(System.lineSeparator())
                     .append(buy).append(",").append(totalBuy).append(System.lineSeparator())
                     .append(resultName).append(",").append(result);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,7 +1,61 @@
...
+        for (String tips : arr) {
+            int intValue = parseInt(tips.substring(tips.indexOf(",") + 1));
+            if (tips.contains("buy")) {
+                totalBuy += intValue;
+            } else if (tips.contains("supply")) {
+                totalSupply += intValue;
+            }
+        }
+        result = totalSupply - totalBuy;
+        builder.append(supply).append(",").append(totalSupply).append(System.lineSeparator());
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:55]: Introduce `","` as constant

**Review by petro-ushchuk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FormattData.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
```
Comment (Andriy-dz) [inline src/main/java/core/basesyntax/FormattData.java:1]: instead of 3 additional private methods, you created 2 classes.

**Review by Andriy-dz — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 9]
```diff
@@ -1,7 +1,71 @@
...
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(managedData))) {
+            for (int i = 0; i < formattedData.length(); i++) {
+                writer.write(formattedData.charAt(i));
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't write data to file " + managedData, exception);
+        }
     }
 }
+
```
Comment (Andriy-dz) [inline src/main/java/core/basesyntax/WorkWithFile.java:70]: empty line

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 10]
```diff
@@ -1,7 +1,71 @@
...
+        StringBuilder builder = new StringBuilder();
+        String value;
+        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
+            while (reader.ready()) {
+                builder.append(reader.readLine()).append(REGEX);
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't read data from file " + readFile, exception);
+        }
```
Comment (Andriy-dz) [inline src/main/java/core/basesyntax/WorkWithFile.java:31]: it should also be a separate method

File: src/main/java/core/basesyntax/FormattData.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+import static java.lang.Integer.parseInt;
+
+public class FormattData extends WorkWithFile {
```
Comment (Andriy-dz) [inline src/main/java/core/basesyntax/FormattData.java:5]: please delete this class

File: src/main/java/core/basesyntax/WriteData.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class WriteData extends WorkWithFile {
```
Comment (Andriy-dz) [inline src/main/java/core/basesyntax/WriteData.java:8]: please delete this class

**Review by Andriy-dz — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 13]
```diff
@@ -1,7 +1,73 @@
...
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
+            while (reader.ready()) {
+                builder.append(reader.readLine()).append(REGEX);
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't read data from file " + readFile, exception);
+        }
+        String formattedData = getFormattedData(builder.toString());
+        writeDataToFile(formattedData, toFileName);
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:35]: Return `builder.toString()` from`readDataFromFile` and use `String formattedData = getFormattedData(...);` `writeDataToFile(formattedData, toFileName);` in `getStatistic` method

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 14]
```diff
@@ -1,7 +1,73 @@
...
+        int result;
+        int totalSupply = 0;
+        int totalBuy = 0;
+        StringBuilder builder = new StringBuilder();
+        String[] arr = data.split(REGEX);
+        for (String tips : arr) {
+            final int intValue = parseInt(tips.substring(tips.indexOf(",") + 1));
+            if (tips.contains("buy")) {
+                totalBuy += intValue;
+            } else if (tips.contains("supply")) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:48]: Make literals constant

File: src/main/java/core/basesyntax/WriteData.java [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+    public static void writeDataToFile(String formattedData, String direction) {
+        File managedData = new File(direction);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(managedData))) {
+            for (int i = 0; i < formattedData.length(); i++) {
+                writer.write(formattedData.charAt(i));
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't write data to file " + managedData, exception);
+        }
+    }
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WriteData.java:18]: No need to create new class for these proposes, move it to `WorkWithFile` class.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 16]
```diff
@@ -1,7 +1,73 @@
...
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    public static final String REGEX = " ";
+    public static final String BUY = "buy";
+    public static final String SUPPLY = "supply";
+    public static final String RESULT_REGEX = ",";
+    public static final String RESULT_NAME = "result";
+
+    public static void getStatistic(String fromFileName, String toFileName) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:19]: `getStatistic` should be non-static

File: src/main/java/core/basesyntax/FormattData.java [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+    public static String getFormattedData(String data) {
+        int result;
+        int totalSupply = 0;
+        int totalBuy = 0;
+        StringBuilder builder = new StringBuilder();
+        String[] arr = data.split(REGEX);
+        for (String tips : arr) {
+            final int intValue = parseInt(tips.substring(tips.indexOf(",") + 1));
+            if (tips.contains("buy")) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/FormattData.java:14]: Make literals constant

File: src/main/java/core/basesyntax/FormattData.java [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+                totalBuy += intValue;
+            } else if (tips.contains("supply")) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/FormattData.java:16]: Make literals constant

File: src/main/java/core/basesyntax/FormattData.java [hunk 19]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import static java.lang.Integer.parseInt;
+
+public class FormattData extends WorkWithFile {
+    public static String getFormattedData(String data) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/FormattData.java:6]: No need to create separate class, move this method to `WorkWithFile`

**Review by petro-ushchuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**